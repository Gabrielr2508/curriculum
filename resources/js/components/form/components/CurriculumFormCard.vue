<template>
	<v-flex xs12 sm6 offset-sm3>
		<v-card>
			<v-card-title primary-title>
			</v-card-title>
			<v-card-text>
				<v-form>
					<v-container>
						<v-layout column>
							<v-flex
								xs12
							>
								<v-text-field
									label="Nome*"
									outline
									prepend-icon="person"
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="E-mail*"
									outline
									prepend-icon="email"
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="Telefone*"
									outline
									prepend-icon="call"
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="Linkedin"
									outline
								>
									<template v-slot:prepend>
											<div class="v-input__icon v-input__icon--prepend">
												<i aria-hidden="true" class="v-icon material-icons icon-linkedin1 theme--light"></i>
											</div>
									</template>
								</v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="GitHub"
									outline
								>
									<template v-slot:prepend>
											<div class="v-input__icon v-input__icon--prepend">
												<i aria-hidden="true" class="v-icon material-icons icon-github theme--light"></i>
											</div>
									</template>
								</v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="filename"
									@click.native="onFocus"
									label="Currículo (pdf)*"
									outline
									prepend-icon="attach_file"
									ref="fileTextField"
								/>
								<input
									:multiple="false"
									:accept="accept"
									@change="onFileChange"
									ref="fileInput"
									class="d-none"
									type="file"
								>
							</v-flex>
							<v-flex xs12>
								<v-select
									v-model="areas"
									:items="options"
									label="Áreas de interesse"
									multiple
									chips
									outline
									deletable-chips
									prepend-icon="work"
									small-chips
								/>
							</v-flex>
							<v-flex
								xs12
							>
								<v-switch
									label="Estágio"
									color="primary"
								/>
							</v-flex>
							<v-flex xs12>
								<v-subheader class="red--text">* campos obrigatórios</v-subheader>
							</v-flex>
						</v-layout>
					</v-container>
				</v-form>
			</v-card-text>
			<v-card-actions>
				<v-btn
					absolute
					dark
					fab
					bottom
					right
					color="primary"
					>
					<v-icon>send</v-icon>
				</v-btn>
			</v-card-actions>
		</v-card>
	</v-flex>
</template>

<script>
export default {
	data() {
		return {
			card_text: 'Test',
			options: [
				'Front-end',
				'Back-end',
				'Administrativo',
				'Financeiro',
				'Cozinha',
				'Limpeza',
				'DevOps',
				'Design',
				'UI/UX'
			],
			areas: [],
			filename: '',
		}
	},

	methods: {
		getFormData(files){
			const data = new FormData();
			[...files].forEach(file => {
				data.append('data', file, file.name); // currently only one file at a time
			});
			return data;
		},

		onFocus(){
			if (!this.disabled) {
				this.$refs.fileInput.click();
			}
		},

		onFileChange($event){
			const files = $event.target.files || $event.dataTransfer.files;
			const form = this.getFormData(files);

			if (files) {
				if (files.length > 0) {
					this.filename = [...files].map(file => file.name).join(', ');
				} else {
					this.filename = null;
				}
			} else {
				this.filename = $event.target.value.split('\\').pop();
			}

			this.$emit('input', this.filename);
			this.$emit('formData', form);
		}
	}
}
</script>

<style>

</style>
