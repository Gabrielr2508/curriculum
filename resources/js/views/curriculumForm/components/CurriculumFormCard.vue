<template>
	<v-flex xs12 sm6 offset-sm3>
		<v-card>
			<v-img
				src="../img/card-bg.jpg"
				aspect-ratio="2.75"
				class="white--text"
			>
				<v-container fill-height fluid>
					<v-layout fill-height align-end>
						<v-flex xs12>
							<span class="headline">May your force be with us</span>
						</v-flex>
					</v-layout>
				</v-container>
			</v-img>

			<v-card-text>
				<v-form
					v-model="valid"
					ref="curriculumForm"
				>
					<v-container>
						<v-layout column>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.name"
									:rules="rules.name"
									label="Nome*"
									outline
									prepend-icon="person"
									required
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.email"
									:rules="rules.email"
									label="E-mail*"
									outline
									prepend-icon="email"
									required
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.phone"
									:rules="rules.phone"
									mask="(##) ##### - ####"
									label="Celular*"
									outline
									required
								>
									<template v-slot:prepend>
											<div class="v-input__icon v-input__icon--prepend">
												<i aria-hidden="true" class="v-icon material-icons icon-whatsapp theme--light"></i>
											</div>
									</template>
								</v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="Linkedin"
									v-model="formData.linkedin"
									:rules="rules.linkedin"
									outline
								>
									<template v-slot:prepend>
											<div class="v-input__icon v-input__icon--prepend">
												<i aria-hidden="true" class="v-icon material-icons icon-linkedin1 theme--light"></i>
											</div>
									</template>
								</v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									label="GitHub"
									v-model="formData.github"
									:rules="rules.github"
									outline
								>
									<template v-slot:prepend>
											<div class="v-input__icon v-input__icon--prepend">
												<i aria-hidden="true" class="v-icon material-icons icon-github theme--light"></i>
											</div>
									</template>
								</v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.filename"
									@click.native="onFocus"
									label="Currículo (pdf)*"
									outline
									prepend-icon="attach_file"
									ref="fileTextField"
									required
									:rules="rules.filename"
								/>
								<input
									:multiple="false"
									accept="application/pdf"
									@change="onFileChange"
									ref="fileInput"
									class="d-none"
									type="file"
								>
							</v-flex>
							<v-flex xs12>
								<v-select
									v-model="formData.areas"
									:items="options"
									:rules="rules.areas"
									label="Áreas de interesse*"
									multiple
									chips
									outline
									deletable-chips
									prepend-icon="work"
									small-chips
									required
								/>
							</v-flex>
							<v-flex
								xs12
							>
								<v-switch
									label="Estágio"
									v-model="formData.is_internship"
									color="primary"
								/>
							</v-flex>
							<v-flex xs12>
								<span class="red--text caption">* campos obrigatórios</span>
							</v-flex>
							<vue-recaptcha
								:sitekey="sitekey"
								ref="invisibleRecaptcha"
								size="invisible"
								badge="bottomleft"
								@verify="onVerifyRecaptcha"
								@expired="resetRecaptcha"
							/>
						</v-layout>
					</v-container>
				</v-form>
			</v-card-text>
			<v-card-actions>
				<v-tooltip top>
					<template v-slot:activator="{ on }">
						<v-btn
							v-on="on"
							@click="handleSubmit"
							absolute
							dark
							fab
							bottom
							right
							color="primary"
						>
							<v-progress-circular
								v-if="disableSubmit"
								indeterminate
							/>
							<v-icon v-else>send</v-icon>
						</v-btn>
					</template>
					<span>Enviar</span>
				</v-tooltip>
			</v-card-actions>
		</v-card>
	</v-flex>
</template>

<script>
import env from '../../../env';
import VueRecaptcha from 'vue-recaptcha';
import validators from '../../../common/validators';

export default {
	components: {
		VueRecaptcha,
	},
	data() {
		return {
			sitekey: env.SITE_KEY,
			options: [
				'Front-end',
				'Back-end',
				'Administrativo',
				'Financeiro',
				'Cozinha',
				'Limpeza',
				'DevOps',
				'Design',
				'UI/UX'
			],
			valid: true,
			disableSubmit: false,
			formData: {
				areas: null,
				filename: '',
				name: '',
				email: '',
				phone: '',
				linkedin: '',
				github: '',
				is_internship: false,
				file: null,
				recaptcha_response: null,
			},
			rules: {
				...validators,
			}
		}
	},

	methods: {
		getFormData(pdf){
			const data = new FormData();
			Object.keys(this.formData)
				.forEach(key => {
					data.append(key, this.formData[key]);
				});

			return data;
		},

		onFocus(){
			if (!this.disabled) {
				this.$refs.fileInput.click();
			}
		},

		onFileChange(){
			const pdf = this.$refs.fileInput.files[0];
			if (pdf) {
				this.formData.filename = pdf.name;
				this.formData.file = pdf;
			} else {
				this.formData.filename = null;
				this.formData.file = null;
			}
		},

		onVerifyRecaptcha(response) {
			this.formData.recaptcha_response = response;

			this.resetRecaptcha();

			console.log('Verify: ' + response)

			axios.post(
				'/curriculum',
				this.getFormData(),
				{
					headers: {
						'Content-Type': 'multipart/form-data'
					}
				}
			)
			.then((response) => {
				console.log(response);
			})
			.catch((error) => {
				console.log(error.response);
			})
			.then(() => {
				this.disableSubmit = false;
			});
		},

		resetRecaptcha() {
			this.$refs.invisibleRecaptcha.reset();
		},

		handleSubmit() {
			if (!this.disableSubmit && this.$refs.curriculumForm.validate()) {
				this.disableSubmit = true;
				this.$refs.invisibleRecaptcha.execute();
			}
		}
	}
}
</script>

<style>

</style>
