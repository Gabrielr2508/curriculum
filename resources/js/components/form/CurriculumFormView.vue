<template>
	<curriculum-form-card class="my-5"/>
</template>

<script>
import CurriculumFormCard from './components/CurriculumFormCard';

export default {
	components: {
		CurriculumFormCard
	},
}
</script>

<style>

</style>
