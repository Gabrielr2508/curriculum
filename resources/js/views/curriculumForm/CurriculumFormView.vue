<template>
	<v-flex xs12>
		<v-toolbar
			app
			color="blue darken-4"
			dark
		>
			<v-toolbar-title>
				<v-img src="../img/logo.png" width="120"></v-img>

			</v-toolbar-title>
			<v-spacer/>

			<v-toolbar-items>
				<v-btn
					flat
					to="/login"
				>
					Login
				</v-btn>
			</v-toolbar-items>
		</v-toolbar>
		<curriculum-form-card class="mb-5 mt-5 pt-5"/>
	</v-flex>
</template>

<script>
import CurriculumFormCard from './components/CurriculumFormCard';

export default {
	components: {
		CurriculumFormCard
	},
}
</script>

<style>

</style>
