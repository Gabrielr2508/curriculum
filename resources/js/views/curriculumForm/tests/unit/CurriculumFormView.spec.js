import { shallowMount } from '@vue/test-utils';
import Vue from 'vue';
import Vuetify from 'vuetify';

import CurriculumFormView from '@/views/curriculumForm/CurriculumFormView';
import CurriculumFormCard from '@/views/curriculumForm/components/CurriculumFormCard';

Vue.use(Vuetify);

describe('CurriculumFormView', () => {
    const build = () => {
        const wrapper = shallowMount(CurriculumFormView);

        return {
            wrapper,
            curriculumFormCard: () => wrapper.find(CurriculumFormCard),
        }
    }

    it('renders the component', () => {
        // arrange
        const { wrapper } = build();

        // assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders main child components', () => {
        // arrange
        const { curriculumFormCard } = build();

        // assert
        expect(curriculumFormCard().exists()).toBe(true);
    });
});
