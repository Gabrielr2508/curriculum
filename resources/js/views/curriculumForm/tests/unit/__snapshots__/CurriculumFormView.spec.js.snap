// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurriculumFormView renders the component 1`] = `
<v-flex-stub tag="div" xs12="">
  <v-toolbar-stub app="true" color="blue darken-4" dark="true" scrollthreshold="300">
    <v-toolbar-title-stub>
      <v-img-stub width="120" src="../img/logo.png" position="center center" transition="fade-transition"></v-img-stub>
    </v-toolbar-title-stub>
    <v-spacer-stub></v-spacer-stub>
    <v-toolbar-items-stub>
      <v-btn-stub activeclass="v-btn--active" tag="button" flat="true" type="button">Login</v-btn-stub>
    </v-toolbar-items-stub>
  </v-toolbar-stub>
  <curriculum-form-card-stub class="mb-5 mt-2"></curriculum-form-card-stub>
</v-flex-stub>
`;
